avro-converter/src/test/java/io/confluent/connect/avro/AvroDataTest.java

```diff
@@ -278,10 +281,133 @@ public void testFromConnectComplex() {
+    org.apache.avro.Schema stringSchema = org.apache.avro.SchemaBuilder.builder().stringType();
+    stringSchema.addProp("connect.doc", "string field");
+    stringSchema.addProp("connect.default", JsonNodeFactory.instance.textNode("foo"));
+    org.apache.avro.Schema bytesSchema = org.apache.avro.SchemaBuilder.builder().bytesType();
+    bytesSchema.addProp("connect.doc", "bytes field");
+    bytesSchema.addProp("connect.default", JsonNodeFactory.instance.binaryNode("foo".getBytes()));
+
+    org.apache.avro.Schema arraySchema = org.apache.avro.SchemaBuilder.builder().array().items().stringType();
+    ArrayNode arrayNode = JsonNodeFactory.instance.arrayNode();
+    arrayNode.add("a");
+    arrayNode.add("b");
+    arrayNode.add("c");
+    arraySchema.addProp("connect.default", arrayNode);
+
+    org.apache.avro.Schema mapSchema = org.apache.avro.SchemaBuilder.builder().map().values().intType();
+    ObjectNode mapNode = JsonNodeFactory.instance.objectNode();
+    mapNode.put("field", 1);
+    mapSchema.addProp("connect.default", mapNode);
+
+    org.apache.avro.Schema nonStringMapSchema = org.apache.avro.SchemaBuilder.builder().array()
+                                                                             .items(complexMapElementSchema);
+    ArrayNode nonStringMapNode = JsonNodeFactory.instance.arrayNode();
+    nonStringMapNode.add(JsonNodeFactory.instance.numberNode(1));
+    nonStringMapNode.add(JsonNodeFactory.instance.numberNode(1));
+    ArrayNode nonStringMapArrayNode = JsonNodeFactory.instance.arrayNode();
+    nonStringMapArrayNode.add(nonStringMapNode);
+    nonStringMapSchema.addProp("connect.default", nonStringMapArrayNode);
+
+    org.apache.avro.Schema avroSchema = org.apache.avro.SchemaBuilder
+        .record(AvroData.DEFAULT_SCHEMA_NAME).namespace(AvroData.NAMESPACE) // default values
+        .fields()
+        .name("int8").doc("int8 field").type(int8Schema).withDefault(2)
+        .name("int16").doc("int16 field").type(int16Schema).withDefault(12)
+        .name("int32").doc("int32 field").type(int32Schema).withDefault(12)
+        .name("int64").doc("int64 field").type(int64Schema).withDefault(12L)
+        .name("float32").doc("float32 field").type(float32Schema).withDefault(12.2f)
+        .name("float64").doc("float64 field").type(float64Schema).withDefault(12.2)
+        .name("boolean").doc("bool field").type(boolSchema).withDefault(true)
+        .name("string").doc("string field").type(stringSchema).withDefault("foo")
+        .name("bytes").doc("bytes field").type(bytesSchema).withDefault(ByteBuffer.wrap("foo".getBytes()))
+        .name("array").type(arraySchema).withDefault(Arrays.asList("a", "b", "c"))
+        .name("map").type(mapSchema).withDefault(Collections.singletonMap("field", 1))
+        .endRecord();
+    org.apache.avro.generic.GenericRecord avroRecord
+        = new org.apache.avro.generic.GenericRecordBuilder(avroSchema)
+        .set("int8", 2)
```

Comment:
These are not defaults. We are constructing an Avro record that we will compare to the generated/converted Avro record, so these need to match the defaults in the schema.

Reply:
I tried using an empty record, but then there are lots of differences between the values added to the expected record and the actual record. For example, the expected Avro record uses `Utf8` instances and `GenericRecord$Array` for values rather than the `String` and `List` as defined on the schema.

I think the current PR is still probably the best compromise.